Makefile

```diff
@@ -1682,3 +1682,12 @@ rustup-install-target-toolchain: rustup-set-version
 .PHONY: changelog
 changelog:
 	@BASE_BRANCH=$(BASE_BRANCH) BASE_TAG=$(BASE_TAG) ./build.assets/changelog.sh
+
+.PHONE: check-gitleaks
+check-gitleaks:
```

Comment:
We might want to pin this to avoid potential issues with new hires/new laptops when gitleaks pushes a breaking change. For example, if gitleaks pushes a v9 that changes the `--staged` flag to something like `--staged-changes`, and somebody installs gitleaks via `brew install gitleaks`, they won't be able to commit anything until this is updated, or they downgrade.

Can we run this as a container image instead (`docker run -v ${path_to_host_folder_to_scan}:/path ghcr.io/gitleaks/gitleaks:latest [COMMAND] --source="/path" [OPTIONS]
`)? This would also remove the need for everybody to explicitly install the tool.

Reply:
Yeah we should pin the version.

Reply:
Hmm I was a bit worried about the runtime in a container because of the overhead on filesystem reads. I tried it when testing and it does take ~2x longer (14ms to 25ms) but it's so fast anyway so probably not a big deal.

Reply:
Pinned the version and changed it to use the container.